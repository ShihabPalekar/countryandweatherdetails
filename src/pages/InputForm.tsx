import { Button, TextField } from "@mui/material";
import { useState } from "react";
import "../styles/inputForm.css";

type Props = {
  navigate: any;
};

const InputForm: React.FC<Props> = ({ ...props }) => {
  const [countryName, setCountryName] = useState("");

  const handleInputChange = (e: { target: { value: string } }) => {
    setCountryName(e.target.value);
  };

  const buttonClicked = () => {
    setCountryName("")
    props.navigate("/info?countryName=" + countryName);
  };

  return (
    <div className="input-form">
      <TextField
        id="standard-basic"
        label="Country name"
        variant="standard"
        onChange={handleInputChange}
        value={countryName}
        role='input-field-country'
      />
      <Button
        disabled={countryName.trim() === ""}
        variant="contained"
        onClick={buttonClicked}
        role="search-button"
      >
        Search
      </Button>
    </div>
  );
};

export default InputForm;