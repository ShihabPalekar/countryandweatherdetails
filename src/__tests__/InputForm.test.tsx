import React from "react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import { fireEvent, render, screen } from "@testing-library/react";
import InputForm from "../pages/InputForm";

test("input form page renders correctly", () => {
  const component = renderer.create(<InputForm navigate={undefined} />);
  let tree = component.toJSON();
  // tree
  expect(tree).toMatchSnapshot();
});

describe("Input Form component", () => {
  test("render input-form component", () => {
    render(<InputForm navigate={undefined} />);
  });

  test("input field for country exists", () => {
    render(<InputForm navigate={undefined} />);
    expect(screen.getByRole("input-field-country")).toBeInTheDocument();
  });

  test("search-button exists", () => {
    render(<InputForm navigate={undefined} />);
    expect(screen.getByRole("search-button")).toBeInTheDocument();
  });

  test("search-button is disabled initially", () => {
    render(<InputForm navigate={undefined} />);
    expect(screen.getByRole("search-button")).toBeDisabled();
  });

});
