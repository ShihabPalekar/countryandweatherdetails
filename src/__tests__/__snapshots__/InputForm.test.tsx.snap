// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input form page renders correctly 1`] = `
<div
  className="input-form"
>
  <div
    className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    role="input-field-country"
  >
    <label
      className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={false}
      htmlFor="standard-basic"
      id="standard-basic-label"
    >
      Country name
    </label>
    <div
      className="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1480iag-MuiInputBase-root-MuiInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
        disabled={false}
        id="standard-basic"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value=""
      />
    </div>
  </div>
  <button
    className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root Mui-disabled  css-sghohy-MuiButtonBase-root-MuiButton-root"
    disabled={true}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="search-button"
    tabIndex={-1}
    type="button"
  >
    Search
  </button>
</div>
`;
